modules/product/src/main/java/com/opengamma/strata/product/fxopt/FxBinaryOption.java

```diff
@@ -0,0 +1,856 @@
+ * The option is European, exercised only on the exercise date.
+ * <p>
+ * For example, a binary call with a strike rate 'EUR 1.00 / USD 1.1' and a payment amount USD 100 pays
+ * this amount at expiry if the future value of the foreign exchange index is at least as great as the
+ * strike rate.
+ */
+@BeanDefinition
+public final class FxBinaryOption
+        implements Product, Resolvable<ResolvedFxBinaryOption>, ImmutableBean, Serializable {
+
+    /**
+     * Whether the option is long or short.
+     * <p>
+     * At expiry, the long party will receive the payment amount in the specified currency if the level of the
+     * foreign exchange index is at or above the strike rate; in this case, the short party will have to deliver
+     * the payment amount in the specified currency to the long party.
+     */
+    @PropertyDefinition(validate = "notNull")
+    private final LongShort longShort;
+    /**
+     * The expiry date of the option.
+     * <p>
+     * The option is European, and can only be exercised on the expiry date.
+     */
+    @PropertyDefinition(validate = "notNull")
+    private final LocalDate expiryDate;
+    /**
+     * The expiry time of the option.
+     * <p>
+     * The expiry time is related to the expiry date and time-zone.
+     */
+    @PropertyDefinition(validate = "notNull")
+    private final LocalTime expiryTime;
+    /**
+     * The time-zone of the expiry time.
+     * <p>
+     * The expiry time-zone is related to the expiry date and time.
+     */
+    @PropertyDefinition(validate = "notNull")
+    private final ZoneId expiryZone;
+    /**
+     * The reference FX Index of the option.
+     * <p>
+     * The reference FX Index is used to determine whether or not a payment event has occurred at the expiry
+     * of the option.
+     */
+    @PropertyDefinition(validate = "notNull")
+    private final FxIndex underlying;
+    /**
+     * The amount and currency of the option payment.
+     * <p>
+     * The payment amount that will be made to the long party in the specified currency, if at expiry, a payment
+     * event has occurred.
+     */
+    @PropertyDefinition(validate = "notNull")
+    private final AdjustablePayment paymentCurrencyAmount;
+    /**
+     * Whether the the option is a put or a call.
+     * <p>
+     * A call on the FX pair is not necessarily the same as a put on the inverse pair - unlike for vanilla
+     * options - since the payment currencies can be different in each case.
+     */
+    @PropertyDefinition(validate = "notNull")
+    private final PutCall putCall;
+    /**
+     * The strike FX rate for the binary option.
+     * <p>
+     * At expiry, the strike rate of the option will be compared to the reference FX Index to determine whether or not
+     * a payment event has occurred.
+     */
+    @PropertyDefinition(validate = "notNull")
+    private final double strike;
```

Comment:
Is it assumed that this strike value is based on `CurrencyPair` in `FxIndex`? If so, add the description to javadoc. IF not, use `FxRate`.